src/libopensc/card-epass2003.c

```diff
@@ -549,49 +693,83 @@ verify_init_key(struct sc_card *card, unsigned char *ran_key, unsigned char key_
 	unsigned char mac[256] = { 0 };
 	unsigned long i;
 	unsigned char tmp_sm;
+	unsigned char isFips;
 	epass2003_exdata *exdata = NULL;
 
 	if (!card->drv_data) 
 		return SC_ERROR_INVALID_ARGUMENTS;
 	exdata = (epass2003_exdata *)card->drv_data;
+	isFips = exdata->bFipsCertification;
 
 	LOG_FUNC_CALLED(card->ctx);
-
-	memcpy(data, ran_key, 8);
-	memcpy(&data[8], g_random, 8);
-	data[16] = 0x80;
-	memset(&data[17], 0x00, blocksize - 1);
-	memset(iv, 0, 16);
+	
+	if(isFips)
+	{
+		memset(data,0x00,15);
+        	memset(&data[11], 0x01, 1);
+        	memset(&data[14], 0x40, 1);
+        	memset(&data[15], 0x01, 1);
```

Comment:
```suggestion
        	data[11] = 0x01;
        	data[14] = 0x40;
        	data[15] = 0x01;
```
this would be more obvious